### PR TITLE
PORTALS-1825, PORTALS-1922

### PR DESCRIPTION
**Pull request by nickgros**
[PORTALS-1825](https://sagebionetworks.jira.com/browse/PORTALS-1825) - Expanding a plot displays the plot in a modal.

[PORTALS-1922](https://sagebionetworks.jira.com/browse/PORTALS-1922) - Pie/Bar plot state no longer resets when selecting a facet. Also tweaked barplot height-width ratio.

https://user-images.githubusercontent.com/17580037/117868751-f4996000-b267-11eb-801c-7c1fafbfa069.mov

### CODE REVIEW

File: src/lib/containers/widgets/facet-nav/FacetNavPanel.tsx [hunk 1]
```diff
@@ -424,100 +406,164 @@ const FacetNavPanel: React.FunctionComponent<FacetNavPanelProps> = ({
...
+        <Modal
+          animation={false}
+          show={showModal}
+          onHide={() => setShowModal(false)}
+        >
+          <Modal.Header closeButton={true}>
+            <Modal.Title>{plotTitle ?? ''}</Modal.Title>
+          </Modal.Header>
+          <Modal.Body>
+            <FacetNavPanel {...props} isModalView={true} />
```
Comment (nickgros) [inline src/lib/containers/widgets/facet-nav/FacetNavPanel.tsx:419]: Kinda strange to call itself, but we can easily recreate the same panel here (with `isModalView` set to true)

File: src/lib/containers/widgets/query-filter/EnumFacetFilter.tsx [hunk 2]
```diff
@@ -308,35 +299,46 @@ export const EnumFacetFilter: React.FunctionComponent<EnumFacetFilterProps> = ({
...
-  // is defined the event occuring is inside the dropdown which we then want to keep open, otherwise
-  // we close it.
-  const onToggle = (_show: boolean, _event: any, metadata: any) => {
-    if (metadata.source) {
-      setIsShowDropdown(true)
-    } else {
-      setIsShowDropdown(false)
-    }
-  }
+  const onToggle = () => setIsShowDropdown(!isShowDropdown)
```
Comment (nickgros) [inline src/lib/containers/widgets/query-filter/EnumFacetFilter.tsx:302]: The old `onToggle` seemed to cause more problems than it solved. I couldn't get the filter menus to close under any circumstances. It was much easier to toggle the open state by clicking the dropdown.

File: src/lib/style/base/_core.scss [hunk 3]
```diff
@@ -1032,7 +1032,7 @@ a.dropdown-item {
   border-top: 1px solid #e9ecef;
 }
 
-.dropdown button.btn {
+.dropdown .btn {
```
Comment (nickgros) [inline src/lib/style/base/_core.scss:1035]: Couldn't override this in a component CSS file, so reduced specificity

File: src/lib/style/bootstrap4_backports/_base-import.scss [hunk 4]
```diff
@@ -98,20 +103,35 @@
...
+      $background: SRC.$gray-light,
+      $border: SRC.$gray-dark,
+      $hover-background: SRC.$gray-light,
+      $active-background: SRC.$gray-light
+    );
+  }
+  .btn-enum-facet-filter-select::after {
+    color: SRC.$secondary-action-color;
+    font-weight: bold;
+  }
```
Comment (nickgros) [inline src/lib/style/bootstrap4_backports/_base-import.scss:136]: New button style for the dropdown, but I think it needs a better name

File: src/lib/style/bootstrap4_backports/_overrides.scss [hunk 5]
```diff
@@ -14,17 +14,20 @@
     }
   }
 
-  // change styling of dropdowns
-  %dropdown-active-state {
-    background: SRC.$primary-action-color;
-    color: white;
```
Comment (nickgros) [inline src/lib/style/bootstrap4_backports/_overrides.scss:20]: This made dropdowns hard to configure. Getting rid of this didn't seem to cause a problem.

File: src/lib/style/bootstrap4_backports/_overrides.scss [hunk 6]
```diff
@@ -38,6 +27,27 @@
...
+
+  button.dropdown-toggle.btn::after {
+    font-size: 18px;
+  }
+
+  .dropdown-toggle.secondary-caret::after {
+    color: SRC.$secondary-action-color;
+    font-weight: bold;
+  }
+
```
Comment (nickgros) [inline src/lib/style/bootstrap4_backports/_overrides.scss:50]: image

**Comment by jay-hodgson** [comment]
Does clicking the "X" in the dialog discard the newly applied filters (that might be the assumption if we have a "Save Filters" button).

File: src/lib/style/bootstrap4_backports/_base-import.scss [hunk 7]
```diff
@@ -54,7 +54,12 @@
 
   // default variant is in BS3 but not BS4, so we recreate it here
   .btn-default {
-    @include button-variant(#fff, #ccc, #e6e6e6, #adadad);
+    @include button-variant(
+      $background: #fff,
+      $border: #ccc,
+      $hover-background: #e6e6e6,
+      $hover-border: #adadad
+    );
```
[the remark this answers is not in the document]
Comment (jay-hodgson) [reply src/lib/style/bootstrap4_backports/_base-import.scss:62]: Thank you!

File: src/__tests__/lib/containers/widgets/facet-nav/FacetNav.test.tsx [hunk 8]
```diff
@@ -97,7 +97,9 @@ describe('facets display hide/show', () => {
   it('when show more is clicked', async () => {
     //const panel = container.querySelector<HTMLElement>('div.FacetNav')
     const button = container.querySelector('button.FacetNav__showMore')
-    fireEvent.click(button!)
+    act(() => {
```
Comment (jay-hodgson) [inline src/__tests__/lib/containers/widgets/facet-nav/FacetNav.test.tsx:100]: Like this, as it should help test reliability.  Out of curiosity, did you actually encounter a timing issue here (with the async button click)?

**Review by jay-hodgson — APPROVED** [review]

File: src/__tests__/lib/containers/widgets/facet-nav/FacetNav.test.tsx (hunk 8 above)
Comment (nickgros) [reply src/__tests__/lib/containers/widgets/facet-nav/FacetNav.test.tsx:100]: Actually, I don't think that this change fixed anything 😕. I think `fireEvent` has `act` builtin, so I'm going to remove it. I was trying to reduce the "not wrapped in `act`" warnings but wasn't successful here.

File: src/lib/containers/query_wrapper_plot_nav/QueryWrapperPlotNav.tsx [hunk 9]
```diff
@@ -110,11 +112,13 @@ const QueryWrapperPlotNav: React.FunctionComponent<QueryWrapperPlotNavProps> = p
...
         <QueryFilterToggleButton />
-        <FacetNav
-          facetsToPlot={facetsToPlot}
-          showNotch={false}
+        <TotalQueryResults
           token={props.token}
+          frontText={''}
+          showNotch={false}
+          {...rest}
         />
```
Comment (nickgros) [inline src/lib/containers/query_wrapper_plot_nav/QueryWrapperPlotNav.tsx:120]: Was in the FacetNav component before, but now pulled out (the two components were separated visually in PORTALS-1820)

File: src/lib/containers/TotalQueryResults.tsx [hunk 10]
```diff
@@ -45,10 +46,9 @@ export type TotalQueryResultsProps = {
 
 const TotalQueryResults: FunctionComponent<TotalQueryResultsProps> = ({
   style,
-  unitDescription,
   frontText,
-  lastQueryRequest,
   token,
+  getLastQueryRequest,
```
Comment (nickgros) [inline src/lib/containers/TotalQueryResults.tsx:51]: `getLastQueryRequest` is supplied by `QueryWrapper`, so just use that

File: src/lib/containers/widgets/facet-nav/FacetNav.tsx [hunk 11]
```diff
@@ -122,18 +121,18 @@ const FacetNav: React.FunctionComponent<FacetNavProps> = ({
...
     ) {
       return 'MORE'
+    } else if (facetUiStateArray.length <= DEFAULT_VISIBLE_FACETS) {
+      return 'NONE'
+    } else {
+      return 'LESS'
     }
-    return 'LESS'
-  }
+  }, [facetUiStateArray])
```
Comment (nickgros) [inline src/lib/containers/widgets/facet-nav/FacetNav.tsx:135]: Before, there was an edge case when there were fewer (or equal) total facets than the default visible.

1. Have, e.g. 1 plotted facet (default visible number of plots is 2)
2. Hide the facet plot
3. You would not see the "Show all plots" button, so the plot could not appear again w/o a refresh.

This fixes it by handling the case where a plot is hidden first. The behavior can be a little strange in some other cases, but not outright broken.

Comment (jay-hodgson) [reply src/lib/containers/widgets/facet-nav/FacetNav.tsx:135]: Ok

**Pull request merged**